app/views/movies/show.html.erb

```diff
@@ -0,0 +1,16 @@
+<h1><%= @movie.title %></h1>
+<h4>Creation Year: <%= @movie.creation_year %></h4>
+<h4>Genre: <%= @movie.genre %></h4>
+<div id="movie-<%= @movie.id %>-actors">
+  <% @actors.order_by_age.each do |actor| %>
+    <h5><%= actor.name %></h5>
+    <h6>Age: <%= actor.age %></h6>
+  <% end %>
+  <h5>Average Age of All Actors: <%= @actors.average_age %></h5>
```

Comment:
Same thing here as above. The thing to be careful of is that you always know what the collection is you're calling this class method on - since it could be either a collection of _all_ actors, or just this movie's actors. Since the user story is asking us for precisely that, (`"I see the average age of all of the movie's actors"`) I would make it an instance method on Movie. 

Your way is not wrong but it can be risky if the implementation is careless (Which yours is not!).